flow/layers/container_layer.h

```diff
@@ -38,6 +47,11 @@ class ContainerLayer : public Layer {
   // For OpacityLayer to restructure to have a single child.
   void ClearChildren() { layers_.clear(); }
 
+  void cacheTotalElevation();
+
+  float elevation_ = 0.0f;
```

Comment:
I wonder if we also need `elevation_` for the leaf layers such as PictureLayer, TextureLayer, etc.?

Reply:
My current understanding is that only `PhysicalLayer` is translated into an entity that needs a z position, so only it really uses elevation. I think the other layers are mostly being used as materials instead in `SceneUpdateContext#SetShapeTextureOrColor`. The only reason I'm hanging these methods off of `ContainerLayer` instead of `PhysicalLayer` is that we need to be able to get the cumulative elevations of the ancestors, which aren't `PhysicalLayer`s here. I could clean this up a little and put all the setter/compute methods just on `PhysicalLayer`, and have it override a getter on `Container`?

Reply:
Good idea: having them only inside `PhysicalShapeLayer` sounds like a much cleaner solution to me! BTW, you can add a `total_elevation` to `struct PrerollContext` and update it during the `Preroll` walk (see, e.g., how we updated the `PrerollContext.cull_rect`), so  `PhysicalShapeLayer` doesn't have to ask its parent for the `total_elevation`. It just uses `context->total_elevation` in `Preroll`. It might be even better to defer the computation all the way back to  `SceneUpdateContext`: add a `total_elevation` field to it and update it while walking the tree in `UpdateScene`.

Reply:
Done. I chose to put this into `PrerollContext`.